library/std/src/net/ip.rs

```diff
@@ -1119,17 +1128,8 @@ impl Ipv6Addr {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn segments(&self) -> [u16; 8] {
-        let arr = &self.inner.s6_addr;
-        [
-            u16::from_be_bytes([arr[0], arr[1]]),
-            u16::from_be_bytes([arr[2], arr[3]]),
-            u16::from_be_bytes([arr[4], arr[5]]),
-            u16::from_be_bytes([arr[6], arr[7]]),
-            u16::from_be_bytes([arr[8], arr[9]]),
-            u16::from_be_bytes([arr[10], arr[11]]),
-            u16::from_be_bytes([arr[12], arr[13]]),
-            u16::from_be_bytes([arr[14], arr[15]]),
-        ]
+        // SAFETY: All element of `s6_addr16` must be big edian.
+        unsafe { Ipv6Bytes { addr: self.inner.s6_addr }.addr16 }
```

Comment:
This will need to do something like `[u16::from_be(arr[0]), u16::from_be(arr[1]), ...]`

Reply:
Thank you. Realized this mistake an hour ago. Will fix it.